typing/typedecl_variance.ml

```diff
@@ -327,7 +327,11 @@ let compute_variance_decl env ~check decl (required, _ as rloc) =
     List.map
       (fun (c, n, i) -> make (not n) (not c) (not abstract || i))
       required
-  | _ -> (* TODO: list every type_kind *)
+  | {type_kind =
+     ( Type_abstract _
+     | Type_record _
+     | Type_variant _
+     | Type_open ); type_manifest = _} ->
```

Comment:
I would propose to split the case in two patterns to make it commutative with the previous case:
```ocaml
| {type_kind = _; type_manifest = Some _ }
| { type_kind = Type_record _ | Type_variant _; type_manifest = _ } ->
```

Reply:
👍